src/main/java/hudson/tasks/junit/SuiteResult.java

```diff
@@ -132,6 +136,154 @@ public SuiteResult(String name, String stdout, String stderr, @CheckForNull Pipe
         this.properties = Collections.emptyMap();
     }
 
+    public SuiteResult(SuiteResult src) {
+        this.file = src.file;
+        this.name = src.name;
+        this.id = src.id;
+        this.duration = src.duration;
+        this.timestamp = src.timestamp;
+        this.time = src.time;
+        this.nodeId = src.nodeId;
+        this.enclosingBlocks = new ArrayList<String>(src.enclosingBlocks);
+        this.enclosingBlockNames = new ArrayList<String>(src.enclosingBlockNames);
+        this.stdout = src.stdout;
+        this.stderr = src.stderr;
+        if (src.cases == null) {
+            this.cases = null;
+        } else {
+            this.cases = new ArrayList<>();
+            for (CaseResult cr : src.cases) {
+                cases.add(new CaseResult(cr));
+            }
+        }
+        this.properties = new HashMap<>();
+        this.properties.putAll(src.properties);
+    }
+
+    public static SuiteResult parse(final XMLEventReader reader, String ver) throws XMLStreamException {
```

Comment:
From what I understand you are manually parsing the saved XMLFile rather than using `XStream` why?

Reply:
There is a huge performance difference. I don't remember the exact details, but from my experience any reflection based serialization is at least 10x slower than manually implemented.